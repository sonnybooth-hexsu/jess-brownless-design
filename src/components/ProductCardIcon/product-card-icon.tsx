import React from "react"
import { Link } from "gatsby"

type ProductCardIconProps = {
  title: string
  path: string
  excerpt: string
  image: string
}

export const ProductCardIcon = ({
  title,
  path,
  excerpt,
  image,
}: ProductCardIconProps) => {
  return (
    <div className="relative flex flex-col px-8 py-12 bg-white border border-black border-solid shadow-sm">
      <img loading="lazy" className="self-start h-12" src={image} alt="Lips" />
      <h2
        data-testid="productCardIconTitle"
        className="mt-8 mb-2 text-3xl tracking-wide lg:text-2xl"
      >
        {title}
      </h2>
      <p className="flex-grow">{excerpt}</p>
      <Link className="absolute top-0 bottom-0 left-0 right-0" to={path}>
        <span className="sr-only">{title}</span>
      </Link>
    </div>
  )
}
