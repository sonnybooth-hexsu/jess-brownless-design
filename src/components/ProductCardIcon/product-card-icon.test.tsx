/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { getByText, getByTestId } from "@testing-library/dom"
import { ProductCardIcon } from "./product-card-icon"

afterEach(() => {
  cleanup()
})

describe("<ProductCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ProductCardIcon
        title="Test title product icon card"
        path="/test-product"
        excerpt="I am a test excerpt"
        image="/test-product.jpg"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the title text passed through", () => {
    const { container } = render(
      <ProductCardIcon
        title="Test title product icon card"
        path="/test-product"
        excerpt="I am a test excerpt"
        image="/test-product.jpg"
      />
    )

    const titleEl = getByTestId(container, "productCardIconTitle")

    expect(getByText(titleEl, "Test title product icon card"))
  })
})
