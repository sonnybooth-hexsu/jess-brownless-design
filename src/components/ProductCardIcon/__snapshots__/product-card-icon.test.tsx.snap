// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
<div>
  <div
    class="relative flex flex-col px-8 py-12 bg-white border border-black border-solid shadow-sm"
  >
    <img
      alt="Lips"
      class="self-start h-12"
      loading="lazy"
      src="/test-product.jpg"
    />
    <h2
      class="mt-8 mb-2 text-3xl tracking-wide lg:text-2xl"
      data-testid="productCardIconTitle"
    >
      Test title product icon card
    </h2>
    <p
      class="flex-grow"
    >
      I am a test excerpt
    </p>
    <a
      class="absolute top-0 bottom-0 left-0 right-0"
      href="/test-product"
    >
      <span
        class="sr-only"
      >
        Test title product icon card
      </span>
    </a>
  </div>
</div>
`;
